src/cmake/common-settings.cmake

```diff
@@ -10,7 +10,7 @@ if (NOT WIN32)
 	set(COMMON_GCC_FLAGS "${COMMON_GCC_FLAGS} -pipe -msse -msse2 -Wunused-but-set-variable -Wunused-but-set-parameter")
 	set(COMMON_GCC_FLAGS "${COMMON_GCC_FLAGS} -Werror=return-type")
 endif()
-set(COMMON_MSVC_FLAGS "/W3 /MP4 /Zi ")
+set(COMMON_MSVC_FLAGS "/W3 /MP4")
```

Comment:
What's /zi option? Why is it removed?

Reply:
See explanation in the PR description:

> Had to remove hard-coded MSVC compiler option /Zi which asked for the generation of debug information, which is anyway not useful in CI.
> In local environment, this will instead be added as necessary by cmake, depending on the type of build requested.

So, actually this hard coded flag did not have anything to do here (actually the other ones would deserve to be questioned too, bu it's not the scopt of the PR).

Reply:
(ccache does not support this flag, probably because it has the effect of producing additional output files, not only compiled code)

Reply:
Indeed, CMake knows how to handle this alone, when building configuration `Debug` or `RelWithDebInfo`.
Setting compiler-specific flags in cmake files should be avoided as much as possible.
The developer is responsible to use the configuration he needs, we should not add debug information for `Release` config.

Reply:
My concern is that by manually setting `COMMON_MSVC_FLAGS`, we might override what was generated by CMake from `CMAKE_BUILD_TYPE`.